llvm/lib/Target/AMDGPU/AMDGPUInstCombineIntrinsic.cpp

```diff
@@ -1024,6 +1024,15 @@ GCNTTIImpl::instCombineIntrinsic(InstCombiner &IC, IntrinsicInst &II) const {
     }
     break;
   }
+  case Intrinsic::amdgcn_wavefrontsize: {
+    // TODO: this is a workaround for the pseudo-generic target one gets with no
+    // specified mcpu, which spoofs its wave size to 64; it should be removed.
+    if ((ST->getCPU().empty() || ST->getCPU() == "generic") &&
```

Comment:
Less than ideal... I am not sure if there is a way to check that a fixed wavefront size is in the subtarget description and not added as an -mattr?

Reply:
None that I could find because we spoof the Wave64 in when it's not specified, so the only differentiator that I could think of is that the `mattr` is never set. If you have an alternative, that'd be great.

Reply:
I do not really have one. Maybe it is OK for now.

Reply:
This is really gross. We also do have a "generic-hsa" target-cpu name

Reply:
Thanks for the input. What is the suggested solution?